ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py

```diff
@@ -22,20 +88,56 @@
 
 
 def create_tracer(otel_tracer):
+    """Creates a :class:`TracerShim` object from the provided OpenTelemetry
+    :class:`opentelemetry.trace.Tracer`.
+
+    The returned :class:`TracerShim` is an implementation of
+    :class:`opentracing.Tracer` using OpenTelemetry under the hood.
+
+    Args:
+        otel_tracer: A :class:`opentelemetry.trace.Tracer` to be used for
```

Comment:
If you're documenting these types anyway, you may want to make them type annotations instead of comments. Sphinx should include these in the generated docs via `sphinx_autodoc_typehints`, which should work even if you're not using the annotations for type checking.

As it is now:

![Screen Shot 2019-10-25 at 2 34 58 PM](https://user-images.githubusercontent.com/43799105/67606098-34a3fb00-f735-11e9-8259-7f2b35b4114c.png)

With a type annotation, using `otel_tracer: "trace.Tracer"` above:

![Screen Shot 2019-10-25 at 2 37 51 PM](https://user-images.githubusercontent.com/43799105/67606104-38d01880-f735-11e9-97fb-1ce8a5b17ccc.png)

One downside is that you have to import the types you use in the annotations.

Reply:
I'm also for type hints, but sphinx also has the `:type:` syntax, and maybe the `Args:` notations also has dedicated support for type docs?

Reply:
Initially I also wanted to use type hints, however the shim package depends heavily on the `opentracing` package which doesn't currently have type annotations, which means we need to write a typing stub for it (or find some alternative solution).

After consulting with @carlosalberto I ended up dropping type annotations altogether as this proved to require a lot of work and we weren't sure about the benefit they would give to the project.

@c24t @Oberon00 do you think we should revisit this?

Reply:
The convention I've seen is to put the type in parens after the arg name, which is what autodoc-typehints does automatically from type hints. Like this:

```
    Args:
        otel_tracer (`opentelemetry.trace.Tracer`): The tracer to be used for
            constructing the `TracerShim`. This tracer will be used to perform
            the actual tracing when user code is instrumented using the
            OpenTracing API.

```

Which should render as:

<img width="721" alt="Screen Shot 2019-11-25 at 12 34 37 PM" src="https://user-images.githubusercontent.com/43799105/69576018-4f5dcf80-0f80-11ea-801c-9a57b080960d.png">

This recommendation comes from the sphinx docs, e.g. the ["google style" docstrings example](https://www.sphinx-doc.org/en/1.7/ext/example_google.html).

> I ended up dropping type annotations altogether

That sounds fine to me, and I agree type annotations can be more trouble than they're worth. I just meant that if you're going to do the work to put the argument types in docstrings, you may as well make them type annotations instead. That way you get the same generated docs (at least as per the convention above), and users who care to use type annotations can use them. This shouldn't require a typing stub for `opentracing`.